cli_composer.go

```diff
@@ -102,7 +102,78 @@ func composeMessageHeader(replyMsg *fbb.Message) *fbb.Message {
 	return msg
 }
 
-func composeMessage(replyMsg *fbb.Message) {
+func composeMessage(args []string) {
+	set := pflag.NewFlagSet("compose", pflag.ExitOnError)
+	subject := set.StringP("subject", "s", "", "")
+	attachments := set.StringArrayP("attachment", "a", nil, "")
+	ccs := set.StringArrayP("cc", "c", nil, "")
+	set.Parse(args)
+	recipients := set.Args() // Call after Parse()
+
+	// Check if any args are set. If so, go non-interactive
+	// Otherwise, interactive
+	if (len(*subject) + len(*attachments) + len(*ccs) + len(recipients)) > 0 {
+		noninteractiveComposeMessage(*subject, *attachments, *ccs, recipients)
+	} else {
+		interactiveComposeMessage(nil)
+	}
+
+}
+
+func noninteractiveComposeMessage(subject string, attachments []string,
+	ccs []string, recipients []string) {
+	// We have to verify the args here. Follow the same pattern as main()
+	// We'll allow a missing recipient if CC is present (or vice versa)
+	if len(recipients)+len(ccs) <= 0 {
+		fmt.Fprint(os.Stderr, "ERROR: Missing recipients in non-interactive mode!\n")
```

Comment:
A Golang idiom that took me a while to learn: lower in the call stack you would want to return an error as part of the function return type. However, this is high enough that handling it here is appropriate (as evidenced by the existing code around it). So good job!

Reply:
Yeah, I was explicitly following the pattern of the surrounding code (and similar as noted in the comment). Happy to take a look at any resources you have on learning idiomatic handling.

Reply:
I learned that particular one from Martin's code reviews, but this appears to be a good source of idioms: https://go.dev/doc/effective_go#errors

Reply:
> Yeah, I was explicitly following the pattern of the surrounding code (and similar as noted in the comment).

You're refactoring some of the first lines of Go I ever wrote IIRC. So probably not the cleanest and idiomatic code I've written 😅